### ISSUE TITLE
bug: k8s ApisixPluginConfig create fail

### ISSUE THREAD

**Issue by wszgrcy** [apache/apisix-ingress-controller#2046]
### Current Behavior

I create ApisixPluginConfig with
```yml
apiVersion: apisix.apache.org/v2
kind: ApisixPluginConfig
metadata:
  name: gzipa
  namespace: default
spec:
  plugins:
    - enable: true
      name: gzip
```
And `apisix-ingress-controller` logs as follow
If I config plugins with apisix dashboard It can work(But it cannot be persisted because it is managed by ingress)

### Expected Behavior

like [https://apisix.apache.org/docs/ingress-controller/concepts/annotations/#using-apisixpluginconfig-resource](https://apisix.apache.org/docs/ingress-controller/concepts/annotations/#using-apisixpluginconfig-resource)

create ApisixPluginConfig and run with Ingress

### Error Logs

```log
2023-11-16T19:44:22+08:00	[31merror[0m	apisix/pluginconfig.go:163	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}

2023-11-16T19:44:22+08:00	[33mwarn[0m	apisix/apisix_plugin_config.go:261	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"default/gzipa","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
W1116 19:44:22.465021       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:22.482976       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:22.495296       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
2023-11-16T19:44:23+08:00	[31merror[0m	apisix/pluginconfig.go:163	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}

2023-11-16T19:44:23+08:00	[33mwarn[0m	apisix/apisix_plugin_config.go:261	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"default/gzipa","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
W1116 19:44:23.484029       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:23.505976       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:23.515677       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
2023-11-16T19:44:23+08:00	[34minfo[0m	gin@v1.8.1/context.go:173	path: /healthz, status: 200, method: GET, query: , ip: 10.0.24.12, user-agent: kube-probe/1.28, errors: , cost: 55.338µs
2023-11-16T19:44:24+08:00	[31merror[0m	apisix/pluginconfig.go:163	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}

2023-11-16T19:44:24+08:00	[33mwarn[0m	apisix/apisix_plugin_config.go:261	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"default/gzipa","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
W1116 19:44:24.497989       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:24.507578       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:24.514072       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
2023-11-16T19:44:25+08:00	[31merror[0m	apisix/pluginconfig.go:163	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}

2023-11-16T19:44:25+08:00	[33mwarn[0m	apisix/apisix_plugin_config.go:261	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"default/gzipa","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
W1116 19:44:25.509738       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:25.526283       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:25.533423       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
2023-11-16T19:44:26+08:00	[31merror[0m	apisix/pluginconfig.go:163	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}

2023-11-16T19:44:26+08:00	[33mwarn[0m	apisix/apisix_plugin_config.go:261	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"default/gzipa","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
W1116 19:44:26.522978       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:26.535246       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:26.545034       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
2023-11-16T19:44:27+08:00	[31merror[0m	apisix/pluginconfig.go:163	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}

2023-11-16T19:44:27+08:00	[33mwarn[0m	apisix/apisix_plugin_config.go:261	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"default/gzipa","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
W1116 19:44:27.540412       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:27.555863       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 19:44:27.564229       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
2023-11-16T19:44:29+08:00	[34minfo[0m	gin@v1.8.1/context.go:173	path: /healthz, status: 200, method: GET, query: , ip: 10.0.24.12, user-agent: kube-probe/1.28, errors: , cost: 54.388µs
```

### Steps to Reproduce

k3s docker hostnetwork
1. create ApisixPluginConfig
2. `kubectl logs  -napisix apisix-ingress-controller-xxxxx`
3. log print
```
2023-11-16T20:06:23+08:00	info	gin@v1.8.1/context.go:173	path: /healthz, status: 200, method: GET, query: , ip: 10.0.24.12, user-agent: kube-probe/1.28, errors: , cost: 34.393µs
2023-11-16T20:06:27+08:00	error	apisix/pluginconfig.go:163	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}

2023-11-16T20:06:27+08:00	warn	apisix/apisix_plugin_config.go:261	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"default/gzipa","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
W1116 20:06:27.863319       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 20:06:27.873298       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
W1116 20:06:27.879048       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
2023-11-16T20:06:29+08:00	info	gin@v1.8.1/context.go:173	path: /healthz, status: 200, method: GET, query: , ip: 10.0.24.12, user-agent: kube-probe/1.28, errors: , cost: 40.703µs

```

### Environment

- APISIX version (run `apisix version`): /usr/local/openresty//luajit/bin/luajit ./apisix/cli/apisix.lua version
3.6.0
- Operating system (run `uname -a`): `Linux VM-24-12-ubuntu 5.15.0-88-generic apache/apisix#98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux /docker 24.0.7`
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): nginx version: openresty/1.21.4.2
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):`{"id":"c061fc56-6a2b-4dfb-9dc8-d4701c7d87b0","hostname":"VM-24-12-ubuntu","boot_time":1700126969,"etcd_version":"unknown","version":"3.6.0"}`
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

#### ApisixPluginConfig
```yml

apiVersion: apisix.apache.org/v2
kind: ApisixPluginConfig
metadata:
  creationTimestamp: '2023-11-16T11:44:22Z'
  generation: 1
  managedFields:
    - apiVersion: apisix.apache.org/v2
      fieldsType: FieldsV1
      fieldsV1:
        f:spec:
          .: {}
          f:plugins: {}
      manager: agent
      operation: Update
      time: '2023-11-16T11:44:22Z'
    - apiVersion: apisix.apache.org/v2
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          .: {}
          f:conditions: {}
      manager: apisix-ingress-controller
      operation: Update
      subresource: status
      time: '2023-11-16T11:46:27Z'
  name: gzipa
  namespace: default
  resourceVersion: '71604388'
  uid: 80f947f1-2f37-45fb-95cb-c25e583e7235
spec:
  plugins:
    - enable: true
      name: gzip
status:
  conditions:
    - message: Sync Successfully
      observedGeneration: 1
      reason: ResourcesSynced
      status: 'True'
      type: ResourcesAvailable

```

`apache/apisix-ingress-controller` is `1.6.0` and `1.7.1`

**Comment by wszgrcy** [comment]
new log print in `1.7.1` 

```
2023-11-17T10:04:47+08:00	error	apisix/pluginconfig.go:139	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}
2023-11-17T02:04:47.558398753Z 
2023-11-17T10:04:47+08:00	error	apisix/apisix_plugin_config.go:206	failed to sync ApisixPluginConfig to apisix	{"error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
2023-11-17T10:04:47+08:00	warn	apisix/apisix_plugin_config.go:282	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"default/gzipa","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
W1117 10:04:47.628880       1 warnings.go:70] unknown field "status.conditions[0].lastTransitionTime"
```

**Comment by wszgrcy** [comment]
worked in apisix `3.5`...

**Comment by jjdiazgarcia** [comment]
We are facing exactly the same issue. After defining a plugin, apisix-ingress-controller is complaining about it
```
apiVersion: apisix.apache.org/v2
kind: ApisixPluginConfig
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"apisix.apache.org/v2","kind":"ApisixPluginConfig","metadata":{"annotations":{},"name":"proxy-rewrite-jero-pluginconfig-test","namespace":"jero-test"},"spec":{"plugins":[{"config":{"host":"devops-pluginconfig-test.jero-test.svc.cluster.local"},"enable":true,"name":"proxy-rewrite"}]}}
  creationTimestamp: "2023-11-17T13:16:31Z"
  generation: 3
  name: proxy-rewrite-jero-pluginconfig-test
  namespace: jero-test
  resourceVersion: "696627044"
  uid: a94e15f4-b53c-48f5-8031-40de2671099a
spec:
  plugins:
  - config:
      host: devops-pluginconfig-test.jero-test.svc.cluster.local
    enable: true
    name: proxy-rewrite
status:
  conditions:
  - message: Sync Successfully
    observedGeneration: 3
    reason: ResourcesSynced
    status: "True"
    type: ResourcesAvailable
```

And error log from apisix-ingress-controller
```
2023-11-23T15:23:58+08:00	error	apisix/pluginconfig.go:163	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}

2023-11-23T15:23:58+08:00	warn	apisix/apisix_plugin_config.go:261	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"jero-test/proxy-rewrite-jero-pluginconfig-test","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
```

apisix version is 3.6.0
apisix-ingress-controller version is 1.6.0

**Comment by 223311** [comment]
> We are facing exactly the same issue. After defining a plugin, apisix-ingress-controller is complaining about it
> 
> ```
> apiVersion: apisix.apache.org/v2
> kind: ApisixPluginConfig
> metadata:
>   annotations:
>     kubectl.kubernetes.io/last-applied-configuration: |
>       {"apiVersion":"apisix.apache.org/v2","kind":"ApisixPluginConfig","metadata":{"annotations":{},"name":"proxy-rewrite-jero-pluginconfig-test","namespace":"jero-test"},"spec":{"plugins":[{"config":{"host":"devops-pluginconfig-test.jero-test.svc.cluster.local"},"enable":true,"name":"proxy-rewrite"}]}}
>   creationTimestamp: "2023-11-17T13:16:31Z"
>   generation: 3
>   name: proxy-rewrite-jero-pluginconfig-test
>   namespace: jero-test
>   resourceVersion: "696627044"
>   uid: a94e15f4-b53c-48f5-8031-40de2671099a
> spec:
>   plugins:
>   - config:
>       host: devops-pluginconfig-test.jero-test.svc.cluster.local
>     enable: true
>     name: proxy-rewrite
> status:
>   conditions:
>   - message: Sync Successfully
>     observedGeneration: 3
>     reason: ResourcesSynced
>     status: "True"
>     type: ResourcesAvailable
> ```
> 
> And error log from apisix-ingress-controller
> 
> ```
> 2023-11-23T15:23:58+08:00	error	apisix/pluginconfig.go:163	failed to create pluginConfig: unexpected status code 400; error message: {"error_msg":"invalid configuration: additional properties forbidden, found name"}
> 
> 2023-11-23T15:23:58+08:00	warn	apisix/apisix_plugin_config.go:261	sync ApisixPluginConfig failed, will retry	{"object": {"Type":1,"Object":{"Key":"jero-test/proxy-rewrite-jero-pluginconfig-test","OldObject":null,"GroupVersion":"apisix.apache.org/v2"},"OldObject":null,"Tombstone":null}, "error": "1 error occurred:\n\t* unexpected status code 400; error message: {\"error_msg\":\"invalid configuration: additional properties forbidden, found name\"}\n\n\n"}
> ```
> 
> apisix version is 3.6.0 apisix-ingress-controller version is 1.6.0

same here using the same version.. getting same issue.

**Comment by monkeyDluffy6017** [comment]
@Revolyssup please help to check

**Comment by Revolyssup** [comment]
@223311 @wszgrcy This has been fixed in 3.7, please upgrade. 
Reference: https://github.com/apache/apisix/commit/d753bfc93a095fde34fea32abd3db17df2e2ed6a

**Comment by Revolyssup** [comment]
I am closing the issue. If you have any other queries, please reopen
